src/redux-reducer-effects.ts

```diff
@@ -1,16 +1,30 @@
-import {flatten as flattenOptions, Some, Option} from 'monapt';
 import {StoreEnhancer, StoreCreator, Reducer, Action} from 'redux';
 
-type EnhancedReducer<State, Task> = <Msg>(state: State, msg: Msg) => [State, Option<Task[]>];
+type StateWithTasks<State,Task> = [State, Task | Task[]];
+
+const hasTasks = <S,T>(r: EnhancedReducerResult<S,T>): r is StateWithTasks<S,T> => r instanceof Array;
```

Comment:
Do we need user defined type guards in TS 2 or can we use control flow based analysis instead?